packages/eslint-plugin-excel-custom-functions/src/rules/utils.ts

```diff
@@ -0,0 +1,256 @@
+import { parse as parsePath } from 'path';
+import {
+  AST_NODE_TYPES,
+  ESLintUtils,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import { isReassignmentTarget, getJsDoc } from 'tsutils';
+import * as ts from 'typescript';
+import { Scope } from '@typescript-eslint/experimental-utils/dist/ts-eslint';
+// import * as metadata from "../data/metadata.json";
+// const metadata: {[key: string]: {comment: Array<string>, attributes: any, properties: Array<any>, methods: Array<any>} } = require("../data/metadata.json");
+let version = "0.0.8";
+// import * as office from "@microsoft/office-js"
+
+export enum OfficeCalls {
+  WRITE = "WRITE",
+  READ = "READ",
+  NOTOFFICE = "NOTOFFICE"
+}
+
+type RequiredParserServices = ReturnType<typeof ESLintUtils.getParserServices>;
+export type Options = unknown[];
+export type MessageIds = string;
+
+const REPO_URL = 'https://github.com/arttarawork/Office-Addin-Scripts';
+
+
+export const createRule = ESLintUtils.RuleCreator(name => {
+  const ruleName = parsePath(name).name;
+
+  return `${REPO_URL}/packages/eslint-plugin-office-custom-functions/blob/v${version}/docs/rules/${ruleName}.md`;
+});
+
+//Code to determine if function has @customfunction tag
+
+export function getCustomFunction(
```

Comment:
Please add typing for this function.  What type is "customFunction"? What's the return type?